File: solutions/swift/high-scores/1/Sources/HighScores/HighScores.swift
struct HighScores {
    let scores: [Int]
    let personalBest: Int
    let latest: Int
    let topThree: [Int]
    init(scores: [Int], personalBest: Int? = nil) {
        self.scores = scores
        self.personalBest = personalBest ?? scores.max()!
        self.latest = scores.last!
        self.topThree = Array(scores.sorted().reversed().prefix(3))
    }
}

File: solutions/lua/high-scores/1/high-scores.lua
return function(scores)
  local s, desc = scores, { table.unpack(scores) }
  table.sort(desc, function(a, b) return a > b end)
  return {
    high_scores = s,
    scores = function() return s end,
    latest = function() return s[#s] end,
    personal_best = function() return desc[1] end,
    personal_top_three = function() return { desc[1], desc[2], desc[3] } end
  }
end
